generate/genfilehcl/genfile_test.go

```diff
@@ -0,0 +1,1327 @@
+						body: hcldoc(
+							expr("something", "tm_max(5, 12, 9)"),
+						),
+					},
+				},
+			},
+		},
+		// {
+		// 	name:  "generate file in stack with multiple files",
+		// 	stack: "/stack",
+		// 	configs: []hclconfig{
+		// 		{
+		// 			path: "/stack",
+		// 			add: globals(
+		// 				str("some_string", "string"),
+		// 				number("some_number", 777),
+		// 				boolean("some_bool", true),
+		// 			),
+		// 		},
+		// 		{
+		// 			path:     "/stack",
+		// 			filename: "test.tm.hcl",
+		// 			add: generateHCL(
+		// 				labels("test.txt"),
+		// 				expr("content", "\"something = global.some_string\""),
+		// 			),
+		// 		},
+		// 		{
+		// 			path:     "/stack",
+		// 			filename: "test2.tm.hcl",
+		// 			add: generateHCL(
+		// 				labels("test2.txt"),
+		// 				expr("content", "\"something = global.some_number\""),
+		// 			),
+		// 		},
+		// 		{
+		// 			path:     "/stack",
+		// 			filename: "test3.tm.hcl",
+		// 			add: generateHCL(
+		// 				labels("test3.txt"),
+		// 				expr("content", "\"something = global.some_bool\""),
+		// 			),
+		// 		},
+		// 	},
+		// 	want: []result{
+		// 		{
+		// 			name: "test.txt",
+		// 			hcl: genFile{
+		// 				origin: "/stack/test.tm.hcl",
+		// 				body: hcldoc(
+		// 					str("something", "string"),
+		// 				),
+		// 			},
+		// 		},
+		// 		{
+		// 			name: "test2.txt",
+		// 			hcl: genFile{
+		// 				origin: "/stack/test2.tm.hcl",
+		// 				body: hcldoc(
+		// 					number("something", 777),
+		// 				),
+		// 			},
+		// 		},
+		// 		{
+		// 			name: "test3.txt",
+		// 			hcl: genFile{
+		// 				origin: "/stack/test3.tm.hcl",
+		// 				body: hcldoc(
+		// 					boolean("something", true),
+		// 				),
+		// 			},
+		// 		},
+		// 	},
+		// },
+	}
+
+	for _, tcase := range tcases {
+		t.Run(tcase.name, func(t *testing.T) {
+			s := sandbox.New(t)
+			stackEntry := s.CreateStack(tcase.stack)
+			stack := stackEntry.Load()
+
+			for _, cfg := range tcase.configs {
+				filename := cfg.filename
+				if filename == "" {
+					filename = config.DefaultFilename
+				}
+				path := filepath.Join(s.RootDir(), cfg.path)
+				test.AppendFile(t, path, filename, cfg.add.String())
+			}
+
+			meta := stack.Meta()
+			globals := s.LoadStackGlobals(meta)
+			res, err := genfilehcl.Load(s.RootDir(), meta, globals)
+			assert.IsError(t, err, tcase.wantErr)
+
+			got := res.GeneratedHCLs()
+
+			for _, res := range tcase.want {
+				gothcl, ok := got[res.name]
+				if !ok {
+					t.Fatalf("want hcl code to be generated for %q but no code was generated for it", res.name)
+				}
+				gotcode := gothcl.String()
+				wantcode := res.hcl.body.String()
+
+				assertHCLEquals(t, gotcode, wantcode)
+				assert.EqualStrings(t,
+					res.hcl.origin,
+					gothcl.Origin(),
+					"wrong origin config path for generated code",
+				)
+
+				delete(got, res.name)
+			}
+
+			assert.EqualInts(t, 0, len(got), "got unexpected exported code: %v", got)
+		})
+	}
+}
+
+func TestPartialEval(t *testing.T) {
```

Comment:
I don't think we have partial evaluation on generate_file, we just fully evaluate contents and then save the evaluated result in a file